### PR TITLE
Instructions Page

### LINKED ISSUE #32: Instructions page or section.
Provide instructions to users on what they are expected to do. E.g. 

```Create a class diagram for the application using UML. You are allowed to use the diagram editor of your choice. You may use the embedded diagrams.net editor. Supported file formats are svg, jpeg, png.```

### PR DESCRIPTION
**Pull request by o-bm**
Created the instructions page. It fetches instructions data (In JSON format) from the back end and render it dynamically to display to the user. 
It is very easy to add or modify the instructions as you can just make changes to the structure in InstructionsController.js. Each instruction has a title and a body that is used to display on frontend. 

Note that there is a previous branch called instructions_page but something got messed up because of an update to main that I needed  so I will delete it. Nothing to worry about there. 

Also I suggest that we maybe use the instructions page as a FAQ. I think this will be better for the application and we can address common issues or questions that users may have while still including the instructions on how to use the app.

close #32

### CODE REVIEW

File: client/src/pages/InstructionsPage.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import React, { useState, useEffect } from 'react';
+import { Container, Accordion, Card } from 'react-bootstrap';
+
+// Define the structure of each instruction
+interface Instruction {
+  title: string;
+  body: string;
+}
+
+const InstructionsPage: React.FC = () => {
```
Comment (arthurhohinng) [inline client/src/pages/InstructionsPage.tsx:10]: React.FC isn't preferred
https://github.com/facebook/create-react-app/pull/8177

**Review by arthurhohinng — COMMENTED** [review]
Could I get an example of when the instructions page is used?
Also an image would be helpful

File: client/src/pages/InstructionsPage.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  const [instructions, setInstructions] = useState<Instruction[]>([]);
+
+  useEffect(() => {
+    // Fetch the instructions from the backend to display
+    fetch('/api/instructions')
```
Comment (Vladmidir) [inline client/src/pages/InstructionsPage.tsx:15]: we can hardcode instructions on the client side.

**Review by Vladmidir — CHANGES_REQUESTED** [review]
I strongly suggest you remove the server components.

Also, the ticket implied that you will write the actual instructions content.

File: server/routes/InstructionsRoutes.js
[the remark this answers is not in the document]
Comment (o-bm) [reply server/routes/InstructionsRoutes.js:1]: Agreed. Will have the instructions hardcoded and make it a react component

File: server/index.js [hunk 3]
```diff
@@ -28,6 +28,8 @@ const challenges = require('./routes/ChallengeRoutes');
 const solutions = require('./routes/SolutionRoutes');
 const users = require('./routes/UserRoutes');
 const comments = require('./routes/CommentRoutes');
+const instructionsRoutes = require('./routes/InstructionsRoutes');
+app.use('/api', instructionsRoutes);
```
[the remark this answers is not in the document]
Comment (o-bm) [reply server/index.js:32]: Thanks for pointing that out

File: client/src/pages/InstructionsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import React, { useState, useEffect } from 'react';
+import { Container, Accordion, Card } from 'react-bootstrap';
+
+// Define the structure of each instruction
+interface Instruction {
```
[the remark this answers is not in the document]
Comment (o-bm) [reply client/src/pages/InstructionsPage.tsx:5]: noted

File: server/controllers/InstructionsController.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+const instructions = [
+    {title: "Getting Started", body: "This is a new instruction"},
```
[the remark this answers is not in the document]
Comment (o-bm) [reply server/controllers/InstructionsController.js:2]: will do

**Comment by o-bm** [comment]
Removed backend integration for  instructions 
Made react component for instructions
Updated challenge page so instructions pop up when a user clicks on a challenge 

image

**Pull request merged**